source/node.h

```diff
@@ -0,0 +1,106 @@
+#include <memory>
+#include <iterator>
+#include <unordered_map>
+#include <any>
+#include <vector>
+
+namespace uast {
+    class Node {
+        std::unordered_multimap<std::string, std::shared_ptr<Node>> nodes_;
+        std::string type_;
+
+    public:
+        class NodeIterator : std::iterator<std::forward_iterator_tag, Node> {
+            std::unordered_multimap<std::string, std::shared_ptr<Node>>::iterator cur_;
+
+        public:
+            NodeIterator(std::unordered_multimap<std::string, std::shared_ptr<Node>>::iterator start) : cur_(start) {
+            }
+
+            virtual NodeIterator operator++(int);
+
+            virtual NodeIterator &operator++();
```

Comment:
I don't know about the style requirements of clang, but as for me it is better to write something like this:
```suggestion
            virtual NodeIterator & operator++();
```
or
```suggestion
            virtual NodeIterator& operator++();
```